---
title: "senegal vs Spain Understory Analysis ITS"
author: "Elena D. DÃ­az Santiago"
date: "2024-10-08"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      message = FALSE,
                      warning = FALSE,
                      comment = "",
                      eval=  TRUE,
                      include = TRUE)

```

```{r }
library(phyloseq)
library(microbiomeMarker)
library(qiime2R)
library(ggplot2)
library(dplyr)
library(magrittr)
library(DESeq2)
library(plotly)
```

```{r echo=FALSE}
#load own functions
source("C:/Users/elenad/Documents/proyectos/scripts/functions.R")
```

In this analysis we want to test Holobiont hypothesis by analysing the genus that are associated to *Maytenus senegalensis* by studing the common genus in understory in both countries, Senegal and Spain.  
  
# Load Data
  
The sequences were analyse with QIIME2 software. We are going to load the different datasets with the help of qiime2R package and build a phyloseq object. To this end we need:  
- Feature table  
- Taxonomy table  
- Rooted tree file  
- Metadata file  

```{r }
physeq <-qza_to_phyloseq(features="taxonomy/len1/final_FeatureTable.qza", 
                         tree="phylogeny/len1/rooted-tree.qza", 
                         taxonomy="taxonomy/len1/taxonomy.qza", 
                         metadata="metadata.txt")
```

## Obtain Understory data
  
The first step include remove control samples:
  
```{r RemoveControl}
physeq@sam_data$SampleID <- row.names(physeq@sam_data)
#remove mock sample
physeq_wo_mock <- subset_samples(physeq, !SampleID %in% c("MockZymo41", "KitControl40"))
#remove ASVs which the abundance == 1
physeq.clean <- prune_taxa(taxa_sums(physeq_wo_mock) > 1, physeq_wo_mock)
```
  
Then, we obtain only Spain samples:  
  
```{r ObtainSpainSamples}
ps.in <- subset_samples(physeq.clean, INorOUT == "Understory")
#remove ASVs with a total abundance 1 in all samples
physeq.in.clean <- prune_taxa(taxa_sums(ps.in) > 1, ps.in)

```

## Normalization

Generally, sequencing data sets are not of the same size. So, to make relatively fair comparisons between sets, we need to normalize the counts. It is necessary to prepare the data in a way that improves the comparability of the samples. There are different methods to perform normalization, but in this case we are using Cumultive Sum Scaling with microbiomeMarker package.

```{r PhyloseqNormalizarion}
#cumulative sum scaling
physeq.css <- normalize(physeq.in.clean, "CSS")
```


# Analysis of the Taxa within 100% of the samples 

We want to analyse the Phyla and Genera which are withing all the samples.

## Count the number of Taxa per experimental conditions

**At Genus level:**

```{r GenusAllSamples}
#aglomerate ASVs at genus evel
ps.genus <- tax_glom(physeq.css, taxrank="Genus")
#remove Genus not in all samples
ps.genus.all_samples <- prune_taxa(rowSums(otu_table(ps.genus) != 0) == ncol(otu_table(ps.genus)), ps.genus)
genus_l <- get_taxa_unique(ps.genus.all_samples, taxonomic.rank="Genus")
#Obtain all the genera
genus_l <- get_taxa_unique(ps.genus.all_samples, taxonomic.rank="Genus")
#remove unasigned genera
genus_to_remove <- c(NA, "uncultured", "Fungi_gen_Incertae_sedis")
genus_l <- genus_l[!genus_l %in% genus_to_remove]
ps.genus.all <- subset_taxa(ps.genus.all_samples, Genus %in% genus_l)

genus_df.all <- psmelt(ps.genus.all)

#Remove abundance 0
genus_df.all <- genus_df.all[!genus_df.all$Abundance == 0,] 
genus_sumarize.all <- genus_df.all %>%  group_by(Site, Genus) %>% 
  summarise(Abundance=mean(Abundance, na.rm=TRUE), sd=sd(Abundance, na.rm=TRUE))

#Print number of Genera
cat("Genera Number= ", length(unique(genus_df.all$Genus)))

```

**At Phylum level:**

```{r PhylumAllSamples}
#agglomerate taxonomic results to Phylum level
ps.phylum <- tax_glom(physeq.css, taxrank="Phylum")
ps.phylum.all <- prune_taxa(rowSums(otu_table(ps.genus) != 0) == ncol(otu_table(ps.genus)), ps.genus)
phylum_df.all <- psmelt(ps.phylum.all)

#Remove abundance 0
phylum_df.all <- phylum_df.all[!phylum_df.all$Abundance == 0,]
phylum_sumarize.all <- phylum_df.all %>%  group_by(Site, Phylum) %>% 
  summarise(Abundance=mean(Abundance, na.rm=TRUE), sd=sd(Abundance, na.rm=TRUE))


cat("Phyla Number= ", length(unique(phylum_df.all$Phylum)))
```

## Composition barplots 

### Heatmap Abundance per sample

With this plot we want to see the microbial composition in all samples and ensure that the ASVs are within all samples in each  experiment group. Moreover, we can see if there is any difference in the abundance of the ASVs in the different experimental conditions.

**At Phylum level**

```{r AbundancePerSampleHeatmapPhylymAllSamples, fig.cap="Heatmap comparing abundance of phyla across experimental groups. Color gradients represents the abundance. Each row corresponds to a phylum, while each column represents a sample grouped by experimental condition."}

ggplotly(plot_abundance_heatmap(phylum_df.all, "Abundance", "Phylum", "Sample") + facet_wrap(~Site, scales = "free_x"))
```

**At Genus level**

```{r AbundancePerSampleHeatmapGenusAllSamples, fig.cap="Heatmap comparing abundance of genera across experimental groups. Color gradients represents the abundance. Each row corresponds to a genus, while each column represents a sample grouped by experimental condition."}

ggplotly(plot_abundance_heatmap(genus_df.all, "Abundance", "Genus", "Sample") + facet_wrap(~Site, scales = "free_x"))
```

### Heatmap Abundance per experimental condition

With this plot we want to see the differences in the abundance of the taxa in the different experiment group.

**At Phylum level**

```{r AbundanceHeatmapPhylymINOUT, fig.cap="Heatmap comparing abundance of phyla across experimental groups. Color gradients represents the abundance. Each row corresponds to a phylum, while each column represents an experimental group, facilitating the comparison of phylum distribution and abundance patterns across groups."}

ggplotly(plot_abundance_heatmap(phylum_sumarize.all, "Abundance", "Phylum", "Site"))
```

**At Genus level**

```{r AbundanceHeatmapGenusINOUT, fig.cap="Heatmap comparing relative abundance of genera across experimental groups. Color gradients represents the abundance. Each row corresponds to a Genus, while each column represents an experimental group, facilitating the comparison of genus distribution and abundance patterns across groups."}

ggplotly(plot_abundance_heatmap(genus_sumarize.all, "Abundance", "Genus", "Site"))
```

We compared the obtained genera with Gap condition from Senegal and Spain:   
  
**Senegal**

There are some genera that are common to Senegal Gap condition, but there are others that only within the samples of Understory condition (Vishniacozyma, Orbiliales, Auxarthron) (see suplementary file III). Moreover, we performed a differential abundance analysis, and found that Chaetomiaceae, Aspergillus, Agaricus, and Cladosporium are more abundant under Understory condition with DESeq2 and Cladosporium with LefSE. Hypocrealles is more abundant in Gap conditon.  
  
**Spain**  
There are some genera that are within the samples of Spain Gap condition, however, Chaetomyaceae, Hypocrealles, Auxarthron, Aspergillius, Ascomycota and Agaricus are only found under Understory condition in both countries. Moreover, Fusarium, Spizellomycetales and Cladosporium genera are differential more abundant under Understory condition in both DESeq2, whereas Cladosporium and Fusarium are found with LefSE (see suplementary file IV).  

**Conclusion**  
Auxarthron genus is found in all samples in Understory condition for both countries. Cladosporium genus is differentialy more abundant in Understory condition in both countries. Chaetomiaceae, Aspergillius and Agaricus are found in Spain Understory, but not in Gap, and, moreover, is more abundant in Senegal Understory than in Gap.  
   
# Analysis of the Taxa within all samples except 1   

We want to analyse the Phyla and Genera which are withing all the samples except one.

## Count the number of Taxa per experimental conditions

**At Genus level:**

```{r Genus-1Samples}
#remove Genus not in all samples
ps.genus.90_samples <- prune_taxa(rowSums(otu_table(ps.genus) != 0) >= ncol(otu_table(ps.genus)) - 1, ps.genus)
genus_l <- get_taxa_unique(ps.genus.90_samples, taxonomic.rank="Genus")
#Obtain all the genera
genus_l <- get_taxa_unique(ps.genus.90_samples, taxonomic.rank="Genus")
#remove unasigned genera
genus_l <- genus_l[!genus_l %in% genus_to_remove]
ps.genus.90 <- subset_taxa(ps.genus.90_samples, Genus %in% genus_l)

genus_df.90 <- psmelt(ps.genus.90)

#Remove abundance 0
genus_df.90 <- genus_df.90[!genus_df.90$Abundance == 0,] 
genus_sumarize.90 <- genus_df.90 %>%  group_by(Site, Genus) %>% 
  summarise(Abundance=mean(Abundance, na.rm=TRUE), sd=sd(Abundance, na.rm=TRUE))

#Print number of Genera
cat("Genera Number= ", length(unique(genus_df.90$Genus)))

```

## Composition barplots 

### Heatmap Abundance per sample

With this plot we want to see the microbial composition in all samples except 1 and ensure that the ASVs are within all samples en each  experiment group. Moreover, we can see if there is any difference in the abundance of the ASVs in the different experimental conditions.

**At Genus level**

```{r AbundancePerSampleHeatmapGenus-1Samples, fig.cap="Heatmap comparing abundance of genera across experimental groups. Color gradients represents the abundance. Each row corresponds to a genus, while each column represents a sample grouped by experimental condition."}

ggplotly(plot_abundance_heatmap(genus_df.90, "Abundance", "Genus", "Sample") + facet_wrap(~Site, scales = "free_x"))
```

### Heatmap Abundance per experimental condition

With this plot we want to see the differences in the abundance of the taxa in the different experiment group.

**At Genus level**

```{r AbundanceHeatmapGenus-1Samples, fig.cap="Heatmap comparing abundance of genera across experimental groups. Color gradients represents the abundance. Each row corresponds to a Genus, while each column represents an experimental group, facilitating the comparison of genus distribution and abundance patterns across groups."}

ggplotly(plot_abundance_heatmap(genus_sumarize.90, "Abundance", "Genus", "Site"))
```

Allowing that the genera can be missed in one sample, we were able to obtain some additional genera. Then, we compared the obtained genera with Gap condition from Senegal and Spain.   

**Senegal**  
There are some genera that are common to Senegal Gap condition, but there are others that only within the samples of Understory condition (Vishniacozyma, Orbiliales, Mortierella, and Auxarthron) (see suplementary file III). Moreover, we performed a differential abundance analysis, and we found some genera that are more abundant under Understory condition with DESeq2 (Aspergillus, Chaetomiaceae, Cladosporium and Agaricus) and with LefSE (Cladosporium).  
  
**Spain**  
There are some genera that are within the samples of Spain Gap condition, however, Stachybotrys, Rhizophlyctis, Onygenales, , Mortierella, Hypocreales, Geastrum, Chaetomiceae, Auxarthron, Aspergillus and Agaricus were found under Understory condition. Moreover, Spizellomycetales, Fusarium, Ascomycota and Cladosporium genera are differential more abundant under Understory condition with DESeq2, whereas Fusarium, Ascomycota and Cladosporium are more abundant with LefSE(see suplementary file IV). 
  
# Conclusion

-The only genera within the samples of Understory condition are Auxarthron and Mortierella.  
-Agaricus and Aspergillius are found only in Spain Understory condition. However, are found in both Senegal Understory and Gap, but is differentialy more abundant under Understory condition.  
- Cladosporium is found in all the samples of all conditions, but is differentialy more abundant under Understory condition in both Senegal and Spain.  
-Vishniacozyma and Orbiliales are found in Senegal Understory, but not in Gap. However, is found in Spain both Understory and Gap condition. They are not very abundant either in Spain or Senegal.  
-Stacybotrys and Rizhophlytics are found under Spain Understory, but not in Gap. However, is found in Senegal both Understory and Gap condition.  
